esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -3417,18 +3499,56 @@ bool CWsDfuEx::doLogicalFileSearch(IEspContext &context, IUserDescriptor* udesc,
     if (!req.getCacheHint_isNull())
         cacheHint = req.getCacheHint();
 
+    unsigned cacheFrom, cacheTo;
+    Owned<IterateFilesCacheElement> cachedITFS = iterateFilesCache->lookup(context, cacheHint);
+    if (cachedITFS && (cachedITFS->cacheFrom <= pageStart+1) && (pageStart+pageSize <= cachedITFS->cacheTo))
+    {
+        cacheFrom = cachedITFS->cacheFrom;
+        cacheTo = cachedITFS->cacheTo;
+    }
+    else
+    {
+        cacheHint = 0;
+        cacheFrom = pageStart + 1;
```

Comment:
If pageStart was e.g. 1000, then cacheFrom = 1001 and the getLogicalFilesSorted call to follow below is going to pass for startOffset, 1000+1-1001, i.e. 0 and retrieve the wrong results..

As commented previously, I don't think a 'from' to the server is way to go.